pkg/controllers/dnspolicy/dnspolicy_healthchecks.go

```diff
@@ -101,58 +103,48 @@ func (r *DNSPolicyReconciler) expectedProbesForGateway(ctx context.Context, gw c
 		log.V(3).Info("DNS Policy has no defined health check")
 		return nil, nil
 	}
+	ipPattern := `\b(?:\d{1,3}\.){3}\d{1,3}\b`
+	re := regexp.MustCompile(ipPattern)
 
-	for _, listener := range gw.Spec.Listeners {
+	for _, address := range gw.Status.Addresses {
+		port := dnsPolicy.Spec.HealthCheck.Port
+		if port == nil {
+			port = &defaultPort
+		}
 
-		log.V(3).Info("getting dnsrecord", "name", dnsRecordName(gw.Name, string(listener.Name)))
-		dnsRecord := &v1alpha1.DNSRecord{}
-		if err := r.Client().Get(ctx, client.ObjectKey{Name: dnsRecordName(gw.Name, string(listener.Name)), Namespace: dnsPolicy.Namespace}, dnsRecord); err != nil {
-			if errors.IsNotFound(err) {
-				continue
-			} else {
-				return nil, err
-			}
+		matches := re.FindAllString(address.Value, -1)
+		if len(matches) != 1 {
+			log.V(3).Info("Found more or less than 1 ip address")
+			continue
 		}
```

Comment:
Why use a regexp here instead of processing the type of Address (IPAddress, kuadrant.io/MultiClusterIPAddress etc..). I assume this doesn't work for Hostname addresses?

Reply:
The idea is strip out he cluster prefix from the Value. The value (ip address or hostname) is used in the probe. you're right in saying that this wont work in it's current implementation for type hostname. My next suggestion would be to split out using the `/` given this line https://github.com/Kuadrant/multicluster-gateway-controller/blob/main/pkg/controllers/gateway/gateway_controller.go#L216 wdyt?

Reply:
I went a head an implemented the above. Let me know if the preference is to change this but both should be handled now.